include/maths.h

```diff
@@ -45,23 +45,46 @@ inline uint64_t inv64_u64(uint64_t p) {
 }
 
 // montgomery multiply
-inline uint64_t mont_mul(uint64_t a, uint64_t b, uint64_t p, uint64_t p_dash) {
-  uint128_t t = (uint128_t)a * b;
-  uint64_t m = (uint64_t)t * p_dash;
-  uint128_t u = t + (uint128_t)m * p;
-  uint64_t res = u >> 64;
-  int64_t maybe = res - p;
-  return maybe < 0 ? res : (uint64_t)maybe;
+// static inline uint64_t mont_mul(uint64_t a, uint64_t b, uint64_t p, uint64_t p_dash) {
+//   uint128_t t = (uint128_t)a * b;
+//   uint64_t m = (uint64_t)t * p_dash;
+//   uint128_t u = t + (uint128_t)m * p;
+//   uint64_t res = u >> 64;
+//   int64_t maybe = res - p;
+//   return maybe < 0 ? res : (uint64_t)maybe;
+// }
+
+static inline uint64_t mont_mul(uint64_t a, uint64_t b, uint64_t p, uint64_t p_dash) {
```

Comment:
Apparently I can mark constraints with `%` to tell the compiler they commute?

https://gcc.gnu.org/onlinedocs/gcc/Modifiers.html

Seems handy as a and b commute. But not sure how to combine this with the fact whichever one ends up as rax also needs to be the output register.